src/types/context/app_context.go

```diff
@@ -33,6 +33,15 @@ func (a *AppContext) GetServiceContextByLocation(location string) *ServiceContex
 	return nil
 }
 
+// GetDependencyServiceData returns the service data for a particular dependency
```

Comment:
Can we make this a little more clear, something like: returns a map from service role to a service's dependency data (or a dependency's service data, whichever makes more sense)